src/main/java/org/apache/maven/plugin/resources/remote/AbstractProcessRemoteResourcesMojo.java

```diff
@@ -0,0 +1,1358 @@
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.artifact.ArtifactType;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.resolution.ArtifactRequest;
+import org.eclipse.aether.resolution.ArtifactResolutionException;
+import org.eclipse.aether.resolution.ArtifactResult;
+import org.eclipse.aether.util.artifact.JavaScopes;
+
+/**
+ * <p>
+ * Pull down resourceBundles containing remote resources and process the resources contained inside. When that is done,
+ * the resources are injected into the current (in-memory) Maven project, making them available to the process-resources
+ * phase.
+ * </p>
+ * <p>
+ * Resources that end in ".vm" are treated as Velocity templates. For those, the ".vm" is stripped off for the final
+ * artifact name and it's fed through Velocity to have properties expanded, conditions processed, etc...
+ * </p>
+ * Resources that don't end in ".vm" are copied "as is".
+ * <p>
+ * This is a support abstract class, with two non-aggregating and aggregating implementations.
+ * </p>
+ */
+public abstract class AbstractProcessRemoteResourcesMojo
+        extends AbstractMojo
+{
+    private static final String TEMPLATE_SUFFIX = ".vm";
+
+    /**
+     * <p>
+     * In cases where a local resource overrides one from a remote resource bundle, that resource should be filtered if
+     * the resource set specifies it. In those cases, this parameter defines the list of delimiters for filterable
+     * expressions. These delimiters are specified in the form 'beginToken*endToken'. If no '*' is given, the delimiter
+     * is assumed to be the same for start and end.
+     * </p>
+     * <p>
+     * So, the default filtering delimiters might be specified as:
+     * </p>
+     *
+     * <pre>
+     * &lt;delimiters&gt;
+     *   &lt;delimiter&gt;${*}&lt;/delimiter&gt;
+     *   &lt;delimiter&gt;@&lt;/delimiter&gt;
+     * &lt;/delimiters&gt;
+     * </pre>
+     * Since the '@' delimiter is the same on both ends, we don't need to specify '@*@' (though we can).
+     *
+     * @since 1.1
+     */
+    @Parameter
+    protected List<String> filterDelimiters;
+
+    /**
+     * @since 1.1
+     */
+    @Parameter( defaultValue = "true" )
+    protected boolean useDefaultFilterDelimiters;
+
+    /**
+     * The character encoding scheme to be applied when filtering resources.
+     */
+    @Parameter( property = "encoding", defaultValue = "${project.build.sourceEncoding}" )
+    protected String encoding;
+
+    /**
+     * The directory where processed resources will be placed for packaging.
+     */
+    @Parameter( defaultValue = "${project.build.directory}/maven-shared-archive-resources" )
+    private File outputDirectory;
+
+    /**
+     * The directory containing extra information appended to the generated resources.
+     */
+    @Parameter( defaultValue = "${basedir}/src/main/appended-resources" )
+    private File appendedResourcesDirectory;
```

Comment:
If this is going to be a new major, consider that we have `.../resouces`, `.../resources-filtered`, shouldn't this be `.../resources-appended`?

Reply:
This is how it is in original code, did not touch this area.